File: python/challenges/miles_per_gallon.py
# Declare variables and prompt user for values
miles_driven = int(input("How many miles have you driven?\n"))
gas_used = int(input("How many gallons of gas did you use?\n"))

# Calculate miles-per-gallon
miles_per_gallon = miles_driven / gas_used

# Display results
print(f"You drove {miles_per_gallon} miles per gallon of gas.")

File: java/challenges/MilesPerGallon.java


import java.util.Scanner;

public class MilesPerGallon
{
  public static void main(String[] args)
  {
    // Declare variables
    Integer milesDriven;
    Integer gasUsed;
    float gasMileage;
    Scanner keyboard = new Scanner(System.in);

    // Prompt user for miles driven
    System.out.println("How many miles have you driven?");
    milesDriven = keyboard.nextInt();

    // Prompt user for gallons of gas used
    System.out.println("How many gallons of gas did you use?");
    gasUsed = keyboard.nextInt();

    // Calculate miles-per-gallon
    gasMileage = (float)milesDriven / (float)gasUsed;

    // Display results
    System.out.println("You drove " + gasMileage + " miles per gallon of gas");
    
    keyboard.close();
  }
}